lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -834,7 +834,9 @@ public function andWhere($where)
         $where = $this->getQueryPart('where');
 
         if ($where instanceof CompositeExpression && $where->getType() === CompositeExpression::TYPE_AND) {
-            $where = $where->with(...$args);
+            if ($args) {
```

Comment:
Please avoid implicit conversion to a boolean. Use `count($args) > 0`.

Reply:
+2 opcodes 😉 Done.

Reply:
-2 seconds to understand the logic 😎